server/auth_server.go

```diff
@@ -624,9 +624,12 @@ func UpgradeCase() error {
 		if err != nil {
 			return err
 		}
+		if p == nil {
+			return fmt.Errorf("UpgradeCase: Could not get the %s auth provider", val)
+		}
 		pSettings, err := readCommonSettings(p.GetProviderSettingList(false))
 		if err != nil {
-			log.Errorf("GetConfig: Error reading DB settings for previous auth providers %v", err)
+			log.Errorf("UpgradeCase: Error reading DB settings for previous auth providers %v", err)
```

Comment:
It is a bad patter to log an error and then return one. Use errors.Wrap() to wrap the error and provide a new error message and just return it.

The general pattern across all languages is: only log the error if you are handling and _not_ allowing it to bubble up further.